ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -359,21 +372,31 @@ private void initializeInstances(List<InstanceConfig> instanceConfigs) throws Ex
     private void updateStateOfReplicas(List<InstanceConfig> instanceConfigs) {
       for (InstanceConfig instanceConfig : instanceConfigs) {
         int schemaVersion = getSchemaVersion(instanceConfig);
-        if (schemaVersion == 0) {
-          AmbryDataNode node = instanceNameToAmbryDataNode.get(instanceConfig.getInstanceName());
-          if (node == null) {
-            logger.info("Dynamic addition of new nodes is not yet supported, ignoring InstanceConfig {}",
-                instanceConfig);
-          } else {
-            HashSet<String> sealedReplicas = new HashSet<>(getSealedReplicas(instanceConfig));
-            HashSet<String> stoppedReplicas = new HashSet<>(getStoppedReplicas(instanceConfig));
-            for (AmbryReplica replica : ambryDataNodeToAmbryReplicas.get(node)) {
-              replica.setSealedState(sealedReplicas.contains(replica.getPartitionId().toPathString()));
-              replica.setStoppedState(stoppedReplicas.contains(replica.getPartitionId().toPathString()));
+        switch (schemaVersion) {
+          case 0:
+            String instanceName = instanceConfig.getInstanceName();
+            long instanceXid = getXid(instanceConfig);
+            AmbryDataNode node = instanceNameToAmbryDataNode.get(instanceName);
+            if (instanceXid <= currentXid) {
+              if (node == null) {
+                logger.info("Dynamic addition of new nodes is not yet supported, ignoring InstanceConfig {}",
+                    instanceConfig);
+              } else {
+                HashSet<String> sealedReplicas = new HashSet<>(getSealedReplicas(instanceConfig));
+                HashSet<String> stoppedReplicas = new HashSet<>(getStoppedReplicas(instanceConfig));
+                for (AmbryReplica replica : ambryDataNodeToAmbryReplicas.get(node)) {
+                  replica.setSealedState(sealedReplicas.contains(replica.getPartitionId().toPathString()));
+                  replica.setStoppedState(stoppedReplicas.contains(replica.getPartitionId().toPathString()));
+                }
+              }
+              break;
```

Comment:
Shouldn't we place `break` right before `default`?

Reply:
Yes, missed it here. Thanks for catching.